text/0000-deprecate-property-lookup-fallback.md

```diff
@@ -0,0 +1,126 @@
+
+Using this information, we will also have the ability to create a codemod that will help migrate templates.
+
+# How We Teach This
+
+`{{this.foo}}` is the way to access the properties on the component class. This also aligns to property access in JavaScript.
+
+Since the `{{this.foo}}` syntax has worked in Ember.Component (which is the only kind of components available today) since the 1.0 series, we are not really in a rush to migrate the community (and the guides, etc) to using the new syntax. In the meantime, this could be viewed as a tool to improve clarity in templates.
+
+While we think writing {{this.foo}} would be a best practice for new code going forward, the community can migrate at its own pace one component at a time. However, once the fallback functionality is eventually removed this will result in a "Helper not found" error.
+
+## Syntax Breakdown
+The follow is a breakdown of the different forms and what they mean.
+
+- `{{@foo}}` is an argument passed to the component
+- `{{this.foo}}` is a property on the component class
+- `{{#with this.foo as |foo|}} {{foo}} {{/with}}` the `{{foo}}` is a local
+- `{{foo}}` is a helper
+
+# Drawbacks
+The largest downside of this proposal is that it makes templates more verbose, causing developers to type a bit more. This will also create a decent amount of deprecation noise, although we feel like tools like [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow) can help mitigate this.
+
+# Alternatives
+This pattern of having programming model constructs to distinguish between of the backing class and arguments passed to the component is not unique to Ember.
+
+## What Other Frameworks Do
+React has used `this.props` to talk about values passed to you and `this.state` to mean data owned by the backing component class since it was released. However, this approach of creating a specific object on the component class to mean "properties available to the template", would likely be even more an invasive change and goes against the mental model that the context for the template is the class.
+
+Vue requires enumeration of `props` passed to a component, but the values in the template suffer from the ambiguity that we are trying to solve.
+
+Angular relies heavily on the dependency injection e.g. `@Input` to enumerate the bindings that were passed to the component and relies heavily on TypeScript to hide or expose values to templating layer with `public` and `private` fields. Like Vue, Angular does not disambiguate.
+
+## Introduce Yet Another Sigil
+We could introduce another sigil to remove ambiguity. This would address the concern about verbosity, however it is now another thing we would have to teach.
```

Comment:
And in light of `this.` having a meaning in JS, and generally being perceived as relatively low-noise to JS developers, it seems worse to  introduce another sigil.